lib/logstash/inputs/mongodb.rb

```diff
@@ -248,7 +265,8 @@ def run(queue)
               last_id_object = Time.at(last_id)
             end
           end
-          cursor = get_cursor_for_collection(@mongodb, collection_name, last_id_object, batch_size)
+
+          cursor = get_cursor_for_collection(@mongodb, collection_name, first_entry_id_object, last_id_object, batch_size)
```

Comment:
I'd rather only pass get_cursor_for_collection the starting point and batch_size and use a separate method as mentioned above to work out what that should be.